### DISCUSSION TITLE
Contact repository owner

### DISCUSSION THREAD

**Discussion by Tonipetrov** [community/community#23358] [New to GitHub]
<p>We have an organization with ~200 people in it and more than 200 repositories (I’m one of the organization admins).<br>
I need to be able to contact those people or at least the reposity owners (preferably by email). When we were inviting them we were also adding their usernames with the corresponding emails to an excel spreadsheet, but lost it.<br>
And now for some users that have usernames that are not their real names we cannot relly know who’s behind that repository… How can we understand the repositry owner’s email address (or contact them in another way)?</p>

**Comment by byrneh** [comment]
<p>Hi <a class="mention" href="/tonipetrov">@Tonipetrov</a> , welcome to the community</p>
<p>Unfortunately there is no direct messaging feature in GitHub that I am aware of<br>
Some options that may reveal a contact are below</p>
<ol>
<li>
<a href="https://api.github.com/users/username" rel="noopener nofollow ugc">https://api.github.com/users/username</a> if they made their mail public on their profile.</li>
<li>
<a href="https://api.github.com/users/username/events" rel="noopener nofollow ugc">https://api.github.com/users/username/events</a> (search the page for “email”), if they used their public email in commits</li>
<li>You could try reaching out to them in an issue in their repository</li>
<li>For a SAML enabled organization (for a linked member not an outside collaborator) you will be able to derive from the linked identity in you IdP. But from your description this will not apply for your case.</li>
</ol>
<p>Good luck.</p>

**Comment by pete4abw** [comment]
<p><a class="mention" href="/byrneh">@byrneh</a></p>
<p>Brilliant! Thanks.</p>
<p>What would happen if you created an issue in your own repo and then <span class="mention">@mentioned</span> the user you want to reach? Would he receive a notice that he was mentioned?</p>

**Comment by byrneh** [comment]
<aside class="quote no-group" data-username="pete4abw" data-post="3" data-topic="166223">
<div class="title">
<div class="quote-controls"></div>
 pete4abw:</div>
<blockquote>
<p>What would happen if you created an issue in your own repo and then <span class="mention">@mentioned</span> the user you want to reach?</p>
</blockquote>
</aside>
<p>I don’t think you can <span class="mention">@mention</span> any GitHub user account who is not related to or a member of the  repository an issue belongs to?<br>
A user has to be listening to notifications and responding for this approach to be of any use as well.</p>

**Comment by Tonipetrov** [comment, marked as answer]
<p>The workaround we came up eventually was:</p>
<ol>
<li>Add all people we couldn’t identify to a new team we created and named “Unidentified”</li>
<li>Wrote a message in GitHub to that team (mentioned it) that since we cannot identify them we will remove them from the oranization unless they send us an email saying something like “This is my github username related to that email address”.</li>
</ol>
<p>Some of them contacted us, the other we cleaned up. I can say that worked in our case.</p>

**Reply by rafiqueullah14** [reply]
Qlc contract address please

**Reply by R-netizen1** [reply]
Iboto

**Comment by ghost** [comment]
<p>There is that “no-reply” email that every GitHub user has. I wonder if anything could be done with that :face_with_raised_eyebrow:</p>

**Reply by Moreh-LeeJunhyeok** [reply]
can you add more explanation for this? what is no-reply email address for specific user?
